apps/src/applab/redux/data.js

```diff
@@ -20,14 +23,23 @@ const DataState = Record({
 const initialState = new DataState();
 
 export default function (state = initialState, action) {
+  let map;
   switch (action.type) {
+    case ADD_TABLE_NAME:
+      return state.set('tableListMap', Object.assign({}, state.tableListMap, {
+        [action.tableName]: true
+      }));
     case CHANGE_VIEW:
       // Discard table data when not viewing a table, so that we don't momentarily
       // show data for the wrong table when we return to the table view.
       if (action.view !== DataView.TABLE) {
         state = state.set('tableData', {});
       }
       return state.set('view', action.view).set('tableName', action.tableName);
+    case DELETE_TABLE_NAME:
+      map = Object.assign({}, state.tableListMap);
+      delete map[action.tableName];
+      return state.set('tableListMap', map);
```

Comment:
This is a bit of an air bubble under the wall paper. Setting a key to undefined won't prevent it from showing up in Object.keys:
```
var obj = {foo: 'bar'}
undefined
obj.foo = undefined
undefined
Object.keys(obj)
["foo"]
```
so this would push a bit of added complexity into the render method of the DataOverview to check that the keys are truthy (not just present). To clean this up a bit anyway, I've added a block around the inside of the case statement, so that I can move the `let` there without eslint complaining.